File: src/main/java/com/scrumwall/service/item/ItemServiceImpl.scala
package com.scrumwall.service.item

import com.scrumwall.domain.Item
import com.scrumwall.util.debug.LogsToLog4J
import com.scrumwall.dao.item.ItemDao
import java.util.List

class ItemServiceImpl extends ItemService with LogsToLog4J {
  var itemDao: ItemDao = _
  def setItemDao(itemDao:ItemDao){
    this.itemDao = itemDao
  }
  def save(item: Item) : Item = {
    debug( "Saveing item with id: " + item.id )
    itemDao save item
  }
  
  def get(id: Int) : Item = {
    debug( "Getting item with id: " + id )
    itemDao get id
  }

  def getItems(columnId: Int) : List[Item] = {
    debug( "Getting items for column: " + columnId )
    itemDao getItems columnId
  }
  
  def getForSprint(sprintId: Int) : List[Item] = {
    debug("fetching items for sprint " + sprintId)
    itemDao getForSprint sprintId
  }
  
  def remove(id: Int) = {
    itemDao remove id
  }
}

File: src/main/java/com/scrumwall/gwt/server/service/ItemServiceImpl.java
package com.scrumwall.gwt.server.service;

import java.util.List;

import org.springframework.beans.factory.annotation.Autowired;

import com.scrumwall.gwt.RpcFacade;
import com.scrumwall.gwt.client.ItemService;
import com.scrumwall.gwt.server.dao.ItemDao;
import com.scrumwall.gwt.shared.ItemDTO;

@RpcFacade
public class ItemServiceImpl implements ItemService{

	@Autowired
	private ItemDao itemDao;

	public List<ItemDTO> getForColumn(int columnId) {
		return itemDao.getForColumn(columnId);
	}
	
	public void save(ItemDTO item) {
		itemDao.save(item);
	}
	
}
